packages/components/src/components/modal/modal.tsx

```diff
@@ -32,10 +32,11 @@ type TModalElement = {
     onUnmount?: () => void;
     portalId?: string;
     renderTitle?: () => React.ReactNode;
+    should_close_on_click_outside?: boolean;
```

Comment:
Do we have any existing logic for closing popup by clicking outside? Popup for Duration/Stake and Barrier (Higher/Lower) are closing like that

Reply:
Yes, Duration/Stake, etc popups close because they have a close button. Currently, the logic in the code is to allow closing on outside if there's a close button on the modal. This modal doesn't have a close button but needs to be closed on outside click, `no idea why`, so I added this property as there might be more unuserfriendly modals like that in the future.